velox/exec/HashTable.h

```diff
@@ -496,6 +496,18 @@ class HashTable : public BaseHashTable {
     setHashMode(mode, numNew);
   }
 
+  void testStoreRowPointer(uint64_t index, uint64_t hash, char* row) {
```

Comment:
These methods look very strange. 

What is the use case that triggers a bug you ran into? Do we need to store 3B+ entries in a single hash table to hit the issue?

Reply:
@mbasmanova : Is there a precedence of using [friends](https://en.cppreference.com/w/cpp/language/friend) syntax with test classes and methods so that we needn't expose such test APIs ?

We do need number of entries to overflow int32_t.

I would've preferred we write a Velox unit test similar to the semi-join query we saw originally. Though your suggestion to use HashTableTest makes the unit test more fine grained I agree.